unison-core/src/Unison/Name.hs

```diff
@@ -87,38 +84,33 @@ toString :: Name -> String
 toString = Text.unpack . toText
 
 isPrefixOf :: Name -> Name -> Bool
-a `isPrefixOf` b = toText a `Text.isPrefixOf` toText b
-
--- stripTextPrefix a.b. a.b.c = Just c
--- stripTextPrefix a.b  a.b.c = Just .c;  you probably don't want to do this
--- stripTextPrefix x.y. a.b.c = Nothing
--- stripTextPrefix "" a.b.c = undefined
-_stripTextPrefix :: Text -> Name -> Maybe Name
-_stripTextPrefix prefix name =
-  Name <$> Text.stripPrefix prefix (toText name)
+Name a `isPrefixOf` Name b = a `Text.isPrefixOf` b
+Name' a `isPrefixOf` Name' b = toList a `NonEmpty.isPrefixOf` b
 
 -- stripNamePrefix a.b  a.b.c = Just c
 -- stripNamePrefix a.b. a.b.c = undefined, "a.b." isn't a valid name IMO
 -- stripNamePrefix x.y  a.b.c = Nothing, x.y isn't a prefix of a.b.c
 -- stripNamePrefix "" a.b.c = undefined, "" isn't a valid name IMO
 -- stripNamePrefix . .Nat = Just Nat
 stripNamePrefix :: Name -> Name -> Maybe Name
-stripNamePrefix prefix name =
+stripNamePrefix prefix@Name{} name@Name{} =
   Name <$> Text.stripPrefix (toText prefix <> mid) (toText name)
   where
   mid = if toText prefix == "." then "" else "."
-
--- a.b.c.d -> d
-stripPrefixes :: Name -> Name
```

Comment:
This was deleted because it's also exported as `unqualified` (and no one was calling this ver)